private/buf/buflsp/buflsp.go

```diff
@@ -81,6 +84,27 @@ func Serve(
 
 // *** PRIVATE ***
 
+// Clients that we are willing to serve without the override flag.
+var knownClients = []string{
+	// VSCode and derivatives use env.appName as the LSP client name. This in
+	// turn is set from nameLong in product.json. See e.g.
+	// https://github.com/microsoft/vscode/blob/e1de2a458dfb770545489daf499131fd328924e7/src/vs/workbench/services/extensions/common/remoteExtensionHost.ts#L216
+
+	// Names from the official VSCode product.
+	"Visual Studio Code",
+	"Visual Studio Code - Insiders",
+
+	// https://github.com/microsoft/vscode/blob/e1de2a458dfb770545489daf499131fd328924e7/product.json#L3
+	"Code - OSS",
+
+	// https://github.com/VSCodium/vscodium/blob/8bcb412ec53f0232e373d9d04b50f804f090ea67/prepare_vscode.sh#L138
+	"VSCodium",
+	"VSCodium - Insiders",
+
+	// https://github.com/neovim/neovim/blob/9a681ad09e2add96d47bf3f39cca8029f3bf09df/runtime/lua/vim/lsp/client.lua#L539
+	"Neovim",
+}
```

Comment:
Could you also consider whitelisting [Helix editor](https://helix-editor.com/) ?

```suggestion
	// https://github.com/helix-editor/helix/blob/cbbeca6c5227e65bebdbe9abbadbd2202ffc1005/helix-lsp/src/client.rs#L682
	"helix"
}
```

Reply:
No matter what we do, we aren't going to try to intentionally break any editor, but our focus is going to be making this language server work with VS Code. This PR just makes it so anyone using a non-VS Code editor has to explicitly understand their editor is not explicitly supported (by explicitly providing a CLI flag). We won't be testing Helix ourselves, so Helix will not be added to this allowlist.